src/tags/models.py

```diff
@@ -1,7 +1,38 @@
+from django.apps import apps
 from django.db import models
+from django.db.models import Sum
 
 from taggit.managers import TaggableManager
 
+from siteconfig.models import SiteConfig
+
+
+def total_xp_by_tags(user, tags: list):
+    """ 
+        Returns user's total xp over active_semester filtered by tags
+
+        expects a list of tags for tags arg (try list(tags))
+    """ 
+    # get models through here to prevent circular imports
+    QuestSubmission = apps.get_model("quest_manager", "QuestSubmission")
+    BadgeAssertion = apps.get_model("badges", "BadgeAssertion")
+
+    # get all objects with xp that is related to user and tag
+    xp_list = [
+        # QuestSubmission xp sum  
+        QuestSubmission.objects.all_approved(user, active_semester_only=True).filter(
+            is_completed=True, 
```

Comment:
I feel like this is a huge improvement from your last version! You also got it into a single operation. Nicely done! Your Django queryset Kung Fu is improving :)

I think `is_completed` filter is uneccessary, since that's already taken care of in `all_approved()`. Can you confirm?

Reply:
nope ``is_completed`` is necessary, ``all_approved()`` doesnt seem to filter for completed quests only approved ones.

Although, there were a bunch of commented out lines that did filter for it